service/dap/server_test.go

```diff
@@ -345,135 +345,163 @@ func TestSetBreakpoint(t *testing.T) {
 	})
 }
 
+var expectFrames = func(t *testing.T, stResp *dap.StackTraceResponse, wantStart int, wantFrames int, wantTotal int, wantLine int) {
+	t.Helper()
+	if stResp.Body.TotalFrames != wantTotal {
+		t.Errorf("\ngot  %#v\nwant TotalFrames=%d", stResp.Body.TotalFrames, wantTotal)
+	}
+	if len(stResp.Body.StackFrames) != wantFrames {
+		t.Errorf("\ngot  len(StackFrames)=%d\nwant %d", len(stResp.Body.StackFrames), wantFrames)
+	} else {
+		for i := 0; i < wantFrames; i++ {
+			if stResp.Body.StackFrames[i].Id != wantStart+i {
+				t.Errorf("\ngot  %#v\nwant Id=%d", stResp.Body.StackFrames[i], wantStart+i)
+			}
+		}
+		if wantFrames > 0 && wantLine > 0 && stResp.Body.StackFrames[0].Line != wantLine {
```

Comment:
Should this be `stResp.Body.StackFrames[i]`?

If `[0]` is intended, please add a comment and move it out of the loop

Reply:
It is out of the loop. It is part of else {}.
Added a comment.

Reply:
Oops, my bad.